bgs/compactor.go

```diff
@@ -131,9 +173,17 @@ func DefaultCompactorOptions() *CompactorOptions {
 		RequeueLimit:      0,
 		RequeueShardCount: 50,
 		RequeueFast:       true,
+		NumWorkers:        2,
 	}
 }
 
+func newRandFromRoot() *rand.Rand {
```

Comment:
this is unnecessary, the default rand is already well seeded

Reply:
this gives us a *Rand that can be owned and accessed lockless by some thread

Reply:
we're not performance constrained on the rand here at all, and even if we were we could just use `rand.New(rand.NewSource(time.Now().UnixNano()))`